src/Application/Actions/Donations/Update.php

```diff
@@ -142,11 +142,37 @@ protected function action(Request $request, Response $response, array $args): Re
                     $retryCount,
                 ));
 
-                \usleep(100_000 * (2 ** $retryCount)); // pause for 0.1, 0.2, 0.4 and then 0.8s before giving up.
+                $microseconds = (int)(100_000 * (2 ** $retryCount)); // pause for 0.1, 0.2, 0.4 and then 0.8s before giving up.
+                \assert($microseconds >= 0);
+                \usleep($microseconds);
                 $retryCount++;
 
                 $this->entityManager->rollback();
             }
+            catch (InvalidRequestException $invalidRequestException) {
+                if (
+                    str_starts_with(
+                        haystack: $invalidRequestException->getMessage(),
+                        needle: "This PaymentIntent's amount could not be updated because it has a status of canceled",
+                    )
+                ) {
+                    $this->logger->warning(sprintf(
+                        'Stripe rejected payment intent update as PI was cancelled, presumably by stripe' .
+                        ' itself very recently. Donation # %s',
+                        $args['donationId'],
```

Comment:
As in the hook receiver, it's good to have UUID as well/instead

Reply:
Thanks, changed.